File: publicq-main/src/PublicQ.Infrastructure/Options/McpApiKeyOptions.cs
namespace PublicQ.Infrastructure.Options;

/// <summary>
/// MCP API Key Options
/// </summary>
public class McpApiKeyOptions
{
    /// <summary>
    /// Enables or disables API Key authentication
    /// </summary>
    public bool Enabled { get; set; } = false;
    
    /// <summary>
    /// A list of API Keys
    /// </summary>
    public IList<ApiKey> Keys { get; set; } = new List<ApiKey>();
}

/// <summary>
/// Represents an API Key
/// </summary>
public class ApiKey
{
    /// <summary>
    /// Key Name
    /// </summary>
    public string Name { get; set; }
    
    /// <summary>
    /// Key Value
    /// </summary>
    public string Key { get; set; }
    
    /// <summary>
    /// Created By
    /// </summary>
    public string CreatedBy { get; set; } = "unknown";
    
    /// <summary>
    /// Valid Until UTC
    /// </summary>
    public DateTime? ValidUntilUtc { get; set; }
    
    /// <summary>
    /// Creation Date UTC
    /// </summary>
    public DateTime CreatedOnUtc { get; set; }
}

File: publicq-main/client/src/models/mcp-api-key-options.ts
/**
 * Represents an API Key
 */
export interface ApiKey {
  /**
   * Key Name
   */
  name: string;
  
  /**
   * Key Value
   */
  key: string;
  
  /**
   * Created By
   */
  createdBy: string;
  
  /**
   * Valid Until UTC
   */
  validUntilUtc: string | null;
  
  /**
   * Creation Date UTC
   */
  createdOnUtc: string;
}

/**
 * MCP API Key Options
 */
export interface McpApiKeyOptions {
  /**
   * Enables or disables API Key authentication
   */
  enabled: boolean;
  
  /**
   * A list of API Keys
   */
  keys: ApiKey[];
}
